### ISSUE TITLE
remove_task_bar not working with multi_monitor

### ISSUE THREAD

**Issue by keepitsane** [TimUntersberger/wwm#85]
I have both remove_task_bar and multi_monitor set to true and I have a triple monitor setup. The task bar is only removed on the primary monitor and not the two secondary monitors when in work mode.

Please let me know what I additional information I can provide.

**Comment by TimUntersberger** [comment]
One thing I'd need your help with is finding the name of the third taskbar.

Currently wwm supports only the first and second monitor (so if your second monitor doesn't work then that's definetly a bug).

The first monitor taskbar is called `Shell_TrayWnd`
The second monitor taskbar is called `Shell_SecondTrayWnd`

Could you please download [WinSpy](http://www.catch22.net/software/winspy) (a tool for inspecting the properties of a window).

After starting WinSpy you will get a window like the following

![image](https://user-images.githubusercontent.com/32014449/88627731-16e13900-d0ad-11ea-90a1-e587189e5544.png)

if you click on the finder tool it will provide a list of currently open windows

![image](https://user-images.githubusercontent.com/32014449/88627849-4a23c800-d0ad-11ea-9070-4245f355ec35.png)

afterwards you can click on the more button to see a list of open windows

![image](https://user-images.githubusercontent.com/32014449/88627885-5871e400-d0ad-11ea-95af-ebc7ccc679a8.png)

usually the first item in the list are the Windows 10 windows and there you will be able to find the taskbar windows.

They will look like in the following image

![image](https://user-images.githubusercontent.com/32014449/88627394-8dca0200-d0ac-11ea-8e9a-043c0d8cb76d.png)

The please tell me the name of the third taskbar.

**Comment by keepitsane** [comment]
I think it might be a bug, because for me only the the task bar on the primary monitor is removed, both the second and third monitors do not get the task bar removed. I think the issue might be is Windows labels all task bars after the first one as SecondaryTrayWnd like you mentioned, even when there is more than two. As seen in the following WinSpy screenshots:

![winspy_1](https://user-images.githubusercontent.com/13246774/88695736-4fded500-d0d0-11ea-85bf-d38a6e1eb15e.png)
![winspy_2](https://user-images.githubusercontent.com/13246774/88695744-51100200-d0d0-11ea-86d1-94f5b18cca16.png)

When I click on the "000701D8 Shell_SecondaryTrayWnd" and click the flash button the task bar on my second monitor flashes, and when I click on "004E0328 Shell_SecondaryTrayWnd" and click the flash button the task bar on my third monitor flashes. Of course clicking on "000100FE Shell_TrayWnd" flashes my primary monitor.

Unplugging one of the monitors and the as you mentioned the removed task bar seems to work as expected when only two monitors are present. 

Basically I think the duplicate naming of the task bar in multi-monitor setups as Shell_SecondaryTrayWnd, seem to be the cause of this bug.

Please let me know if there any additional testing and screenshots you would like me to do.

**Comment by TimUntersberger** [comment]
I think this will be easier to fix than I initially thought. Everything below this line are notes for myself.

Use [EnumWindows](https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-enumwindows) to enumerate all windows and filter for windows that contain the pattern `^Shell_(Secondary)?TrayWnd$` instead of using hardcoded strings. To find out which monitor holds the window you can use the [MonitorFromWndow](https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-monitorfromwindow) function.

**Comment by TimUntersberger** [comment]
**Thank you for your help!**

**Comment by TimUntersberger** [comment]
This should be fixed in the latest dev build.

**Comment by keepitsane** [comment]
So far this seems to fix the issue. Thanks!

When you officially merge these changes into the release version you should make a note that the monitor numbering scheme has changed. People might be confused when their workspaces  seem to not be mapped correctly. As before this build my monitor number 1 was not my primary monitor, but rather one of my secondary. In this build my primary monitor is number 1 which is the correct way in my opinion.

**Comment by keepitsane** [comment]
I think I closed this issue a bit too soon. It seems the task bar being removed on all monitors only works sometimes, then it gets stuck after sometime and it will either remove the task bar only on the primary monitor or only remove it on the secondary monitor. The only fix I have seemed to find is a complete system restart, even restarting wwm or toggling the remove_task_bar setting doesn't seem to fix the issue.

**Comment by TimUntersberger** [comment]
One thing I noticed is that this can be fixed by also restarting wwm (at least on my machine), so I think it has to be some form of a race condition. I'll try to look at the logs when this happens the next time for me.

**Comment by TimUntersberger** [comment]
I fixed the problem on my own computer (The taskbar on both monitors now become hidden 100% percent of the time), but I never experienced wwm being "stuck". Can you explain a bit more what you mean by this or whether the newest dev build fixed the problem for you?

**Comment by keepitsane** [comment]
Yeah I have been trying to reproduce the issue and can't seem to isolate it. I'll test out the latest build and see if that helps. For now I close this issue again until I can reproduce the bug reliably, because like you said for the most part restarting wwm seems to fix it most of the time. Thanks again!
